libtenzir/src/execution_node.cpp

```diff
@@ -93,6 +94,32 @@ auto make_timer_guard(Duration&... elapsed) {
     });
 }
 
+// Return an underestimate for the total number of referenced bytes for a vector
+// of table slices, excluding the schema and disregarding any overlap or custom
+// information from extension types.
+auto num_approx_bytes(const std::vector<table_slice>& events) {
+#if ARROW_VERSION_MAJOR < 12
+  // arrow::util::ReferenceBufferSize is broken for Arrow before 12.0.0 and
+  // crashes for extension types.
+  (void)events;
+  return uint64_t{0};
+#else
+  auto result = uint64_t{};
+  for (const auto& batch : events) {
+    if (batch.rows() == 0)
+      continue;
+    auto record_batch = to_record_batch(batch);
+    TENZIR_ASSERT(record_batch);
+    // Note that this function can sometimes fail. Because we ultimately want to
+    // return an underestimate for the value of bytes, we silently fall back to
+    // a value of zero is the referenced buffer size cannot be measured.
```

Comment:
Warning/debug message for this?

Reply:
```suggestion
    // Note that this function can sometimes fail. Because we ultimately want to
    // return an underestimate for the value of bytes, we silently fall back to
    // a value of zero if the referenced buffer size cannot be measured.
```

Also this :point_up: